sunpy/net/hek/hek.py

```diff
@@ -96,6 +113,125 @@ def _parse_times(table):
                 table[tkey].format = 'iso'
         return table
 
+    @staticmethod
+    def _parse_unit(table, attribute, is_coord_prop = False):
+        if "is_chaincode" in attribute and attribute["is_chaincode"]:
+            return table
+        unit_attr=""
+        if is_coord_prop:
+            unit_attr = "event_coordunit"
+        else:
+            unit_attr = attribute["unit_prop"]
+        for row in table:
+            if unit_attr in table.colnames and row[unit_attr] not in ["", None] and table[attribute["name"]].unit is not None:
+                table[attribute["name"]].unit = HEKClient._get_unit(attribute["unit_prop"], row[unit_attr], is_coord_prop= is_coord_prop)
+                break
+        return table
+
+    @staticmethod
+    def _parse_astropy_unit(str):
+        try:
+            unit = u.Unit(str)
+        except ValueError:
+            try:
+                unit = u.Unit(str.lower())
+            except ValueError:
+                try:
+                    unit = u.Unit(str.capitalize())
+                except ValueError:
+                    units = str.split(" per ")
+                    unit = None
+                    if len(units) >1:
+                        unit = HEKClient._parse_astropy_unit(units[0])
```

Comment:
Does this method need to be part of the HEKClient?

Reply:
I wonder actually if any of these need to be attached to the client.

Reply:
where can be move them? Actually I found a utility function called `_parse_times` inside the client only so I wrote all the others here only.

Reply:
We move them into a seperate utils.py file? At least outside of the client itself.